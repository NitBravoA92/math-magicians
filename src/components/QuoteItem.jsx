import useQuote from '../hooks/useQuote';

const QuoteItem = () => {
  const { loading, data, error } = useQuote({});

  return (
    <div id="display-quote">
      <p id="quote-text" className={`${(JSON.stringify(data) === '{}' || data === null) ? 'd-none' : ''}`}>
        <span id="quote">{data?.quote || ''}</span>
        <span id="author">
          {' â€” '}
          {data?.author || ''}
        </span>
      </p>
      <p id="loading-message" className={`${!loading ? 'd-none' : ''}`}>{ loading ? 'Loading Quote...' : '' }</p>
      <p id="error-message" className={`${!error ? 'd-none' : ''}`}>{ error || '' }</p>
    </div>
  );
};

export default QuoteItem;
