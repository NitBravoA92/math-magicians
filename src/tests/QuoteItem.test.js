import React from "react";
import '@testing-library/jest-dom/extend-expect';
import { render } from "@testing-library/react";
import renderer from 'react-test-renderer';
import QuoteItem from "../components/QuoteItem";

describe("The QuoteItem component", () => {

  test("renders correctly into the DOM", () => {
    const component = renderer.create(<QuoteItem />).toJSON();

    expect(component).toMatchSnapshot();
  });

  test("display the message 'Loading Quote...' when the API request is processing", () => {
    const component = render(<QuoteItem />);

    const loadingMessageParagraph = component.container.querySelector("#loading-message");

    expect(loadingMessageParagraph).toBeInTheDocument();
    expect(loadingMessageParagraph).toHaveTextContent("Loading Quote...");
  });

});
