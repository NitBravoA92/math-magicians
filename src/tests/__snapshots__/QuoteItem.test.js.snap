// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The QuoteItem component renders correctly into the DOM 1`] = `
<div
  id="display-quote"
>
  <p
    className="d-none"
    id="quote-text"
  >
    <span
      id="quote"
    />
    <span
      id="author"
    >
       â€” 
    </span>
  </p>
  <p
    className=""
    id="loading-message"
  >
    Loading Quote...
  </p>
  <p
    className="d-none"
    id="error-message"
  />
</div>
`;
